docs/source/extension/extension_dual_compatibility.rst

```diff
@@ -0,0 +1,107 @@
+.. Copyright (c) Jupyter Development Team.
+.. Distributed under the terms of the Modified BSD License.
+
+Building Lab 4 / Notebook 7 Compatible Jupyter Extensions
+
+Jupyter Notebook 7 is built with components from JupyterLab, and since
+both use the same building blocks, that means your extension can work
+on both with little or no modification depending on its design.
+
+This guide will give you an overview of compatibility features, then a
+tutorial and reference code covering some of the topics mentioned here.
+If you don't know how to make extensions, you can read more about the
+basics at :ref:`the extensions page <extension_dev>`.
+
+How Compatibility Works
+
+At a high level, extensions for JupyterLab and Jupyter Notebook both
+typically start from a template project. You can download and start modifying
+a template project with `Cookiecutter <https://cookiecutter.readthedocs.io/en/stable/README.html>`_.
+Once your template is ready, you can start adding components and features to
+build your extension.
+
+An extension for Lab (and for Notebook 7) is made up of a `series <https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html>`_
+of bundled `plugins <https://lumino.readthedocs.io/en/latest/api/interfaces/application.IPlugin.html#requires>`_, and those plugins typically use components from the
+interface toolkit `Lumino <https://lumino.readthedocs.io/en/latest/api/index.html>`_
+as well as the `JupyterLab API <https://jupyterlab.readthedocs.io/en/latest/api/index.html>`_
+(among others) to help build your extension's look and behavior (both are
+written in Typescript).
+
+This is how basic compatibility features work: both apps use the same building
+blocks and methods. For instance, both Lab and Notebook 7 accept Lumino widgets
```

Comment:
To disambiguate and for consistency we'll probably want to stick to "JupyterLab" instead of "Lab" in this document.

```suggestion
blocks and methods. For instance, both JupyterLab and Notebook 7 accept Lumino widgets
```